packages/api/src/models/builders/SortQueryBuilder.ts

```diff
@@ -0,0 +1,61 @@
+class SortQueryParam {
+	key: string
+	direction: string
+
+	constructor(key: string, direction: string) {
+		this.key = key
+		this.direction = direction
+	}
+}
+
+/**
+ * Builder to build AxiosRequestConfig 'sort' params
+ * for accounting endpoints
+ */
+export class SortQueryBuilder {
+	queryParams: SortQueryParam[]
+
+	constructor() {
+		this.queryParams = [] as SortQueryParam[]
+	}
+
+	asc(key: string): SortQueryBuilder {
+		this.queryParams.push(new SortQueryParam(key, 'asc'))
+		return this
+	}
+
+	ascending(key: string): SortQueryBuilder {
+		return this.asc(key)
+	}
+
+	desc(key: string): SortQueryBuilder {
+		this.queryParams.push(new SortQueryParam(key, 'desc'))
+		return this
+	}
+
+	descending(key: string): SortQueryBuilder {
+		return this.desc(key)
+	}
+
+	build(resourceType?: string): string {
+		let isAccountingLike = false
+		if (!resourceType || ['AccountingResource', 'EventsResource'].includes(resourceType)) {
+			isAccountingLike = true
+		}
+		let queryString = ''
+		this.queryParams.forEach((param) => {
```

Comment:
Hi. It looks like the API only supports a single `sort` parameter, all others are ignored.
So I think this can be changed to just a single `queryParam` value that gets set with any `asc` or `desc` calls and no need to iterate.